src/scenic/syntax/veneer.py

```diff
@@ -383,15 +387,21 @@ def require(reqID, req, line, name, prob=1):
 	if evaluatingRequirement:
 		raise RuntimeParseError('tried to create a requirement inside a requirement')
 	if currentSimulation is not None:	# requirement being evaluated at runtime
-		if prob >= 1 or random.random() <= prob:
-			result = req()
-			assert not needsSampling(result)
-			if needsLazyEvaluation(result):
-				raise RuntimeParseError(f'requirement on line {line} uses value'
-										' undefined outside of object definition')
-			if not result:
-				raise RejectSimulationException(name)
+		if req.has_temporal_operator:
+			# support monitors on dynamic requirements and create dynamic requirements
+			currentScenario._addDynamicRequirement(requirements.RequirementType.require, req, line, name)
+		else:
+			if prob >= 1 or random.random() <= prob:
+				result = req.evaluate()
+				assert not needsSampling(result)
+				if needsLazyEvaluation(result):
+					raise RuntimeParseError(f'requirement on line {line} uses value'
+											' undefined outside of object definition')
+				if not result:
+					raise RejectSimulationException(name)
 	else:	# requirement being defined at compile time
+		if req.is_temporal and prob != 1:
```

Comment:
This should be `req.has_temporal_operator` to catch temporal operators nested under non-temporal operators, right?

Reply:
Also, shouldn't this check be done in the runtime case as well? Maybe you can just move it to before line 385.